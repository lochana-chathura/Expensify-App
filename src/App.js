import './App.css';
import AppRouter from './routers/AppRouter';
import configureStore from './store/configureStore';
import { addExpense } from './actions/expenses';
import { setTextFilter } from './actions/filters';
import getVisibleExpenses from './selectors/expenses';
import React from "react";
import {Provider} from 'react-redux'

const store=configureStore();

store.dispatch(addExpense({ description: 'Water bill', amount:4500 }));
store.dispatch(addExpense({ description: 'Gas bill', amount:13000 }));
store.dispatch(setTextFilter('water'));

setTimeout(()=>{
    store.dispatch(setTextFilter('bill'));
},3000);
const state = store.getState();
const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
console.log(visibleExpenses);

const jsx=(
    <Provider store={store}>
    <AppRouter/>
    </Provider>
);

export default jsx;
