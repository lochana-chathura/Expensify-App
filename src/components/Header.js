import React from 'react';
import {NavLink} from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Epensify</h1>r
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}>Create Expense</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink>
    </header>
);

export default Header;