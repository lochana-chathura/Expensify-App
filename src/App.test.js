import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import './actions/expenses';
import { addExpense } from './actions/expenses';
import Header from './components/Header';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

test('header test',()=>{
  const wrapper= shallow(<Header/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

// const add=(a,b)=>a+b;
// test('add test',()=>{
//   const result=add(4,3);
//   expect(result).toBe(7);
// });