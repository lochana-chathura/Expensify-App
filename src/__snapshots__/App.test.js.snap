// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test 1`] = `
<header>
  <h1>
    Epensify
  </h1>
  r
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
